services/py-api/src/database/repository/teams_repository.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from bson import ObjectId
+from motor.motor_asyncio import AsyncIOMotorClientSession
+from pydantic import BaseModel
+from pymongo.errors import DuplicateKeyError
+from result import Result, Err, Ok
+from structlog.stdlib import get_logger
+
+from src.database.db_manager import DatabaseManager
+from src.database.model.team_model import Team
+from src.database.repository.base_repository import CRUDRepository
+from src.server.exception import DuplicateTeamName
+from src.server.schemas.request_schemas.schemas import ParticipantRequestBody
+
+LOG = get_logger()
+
+
+class TeamsRepository(CRUDRepository):
+    def __init__(self, db_manager: DatabaseManager, collection: str) -> None:
+        self._client = db_manager.client
+        self._collection = self._client[db_manager.DB_NAME][collection]
+
+    async def create(
+        self, input_data: ParticipantRequestBody, session: Optional[AsyncIOMotorClientSession] = None
+    ) -> Ok[Team] | Err[DuplicateKeyError] | Err[Exception]:
```

Comment:
`Err[DuplicateKeyError | Exception]`

Reply:
Or better yet, define a base error which you can use as the base type and return errors which extend it

Reply:
Fixed in 1a00dd8c6f9016c7770fd6e17ffc6b8195dc8ce2